### DISCUSSION TITLE
AOV output

### DISCUSSION THREAD

**Discussion by POPREQ** [bnpr/Malt#91] [Feedback & Feature Requests]
Hi.

I am very excited about the daily progress of Malt.
As long as AOV output works, I'm looking forward to running Malt in a my project soon.

Do you have any plans to include AOV output or the ability to output multiple paths, and if so, do you have any idea when that might be?

**Comment by BlenderNPR** [comment]
Me, Light, discussed a bit about that with @beesDotMoe few hours ago.
No concrete plan yet but it is on the radar. 
We need solid use cases to design it better. 

If you have common use cases, please list them. 
We can work from there.

**Reply by POPREQ** [reply]
By "common use cases", do you mean how the AOV path is actually used?
Or does it mean what I think would be the ideal form?

I find Eevee's AOV easy to use.
1.You can create an unlimited number of AOV paths.
2. Easy to add, change and check them.
3. render further adjustments to those passes (Blender Composit Node)
4. Render together with the main image and the depth pass at the same time with a single button.
All these elements would be great.


If you have a question about actual use cases, please let us know.
We will post again with the images included and ready to go.

**Reply by BlenderNPR** [reply]
OK, good. The more use cases, the better the design will be for BEER.

The other day on BNPR discord, we discussed about how AOV in UI form for BEER.
Implementing AOV isn't hard since BEER layer stack is designed like AOVs, the hard part is how to present the UI that covers use cases as inclusive as possible.

There are 2 main workflow with AOV that we identified.
1. Working with an external pipeline
2. Working inside Blender for the compositor

Some people use it like render passes, some use it to tweak shading, some use it as base layer to be painted over because painting outside Blender is so much easier.

These AOVs can be in
1. a layer/shader
2. a group of layers/shaders
3. shader container, the material
4. world space AOVs (atmosphere, sky, stars)
5. screen space AOV, only viewable from the camera

These are where we can expose AOVs.

My plan is to find more use cases available for the BEER UI before coding starts.

**Reply by Psi25** [reply]
AOVs would be very helpful for additional tweaking for post-processing using other softwares like After Effects. If AOVs can also be used inside shaders, it can be used for additional control depending on the artstyle.

**Reply by pragma37** [reply]
@Saiyonara25 AOVs are already supported.
Custom outputs from the Render nodes are available in the compositor.

https://malt3d.com/Documentation/Graphs/#custom-io
https://youtu.be/HiGYx_DADOQ

**Reply by Psi25** [reply]
oh

**Comment by pragma37** [comment]
> As long as AOV output works, I'm looking forward to running Malt in a my project soon.

That's awesome!
The immediate plan is to add extra outputs to *Mesh Shaders*, just like in *Screen Shaders* where you have *Output_0*, *Output_1*...
Then you have those outputs available from *Render Layer Input* for post-processing.

If you need the AOVs for compositing them outside Malt (Blender Compositing or other software) that would require more work.

**Reply by pragma37** [reply]
I added the extra outputs to *Mesh Shaders*.

I hope that helps!

**Reply by POPREQ** [reply]
Thank you.

The Screen shader and Render Layer system is magical.
I've been waiting for Eevee's real time compositing and I didn't expect Malt to be the first to make it happen.

But for the time being I'm not going to move away from my general video production style. (DCC + After Effects compositing)

The ability to output multiple images or passes with a single render button makes a huge difference in efficiency.
If I can do that with the ease of Eevee's AOV Output, I'll be very happy indeed. I'm looking forward to seeing it added.

((Also, I haven't had a chance to test this version yet...))
https://www.patreon.com/posts/nodes-are-now-53026781
